nailgun/entities.py

```diff
@@ -7828,6 +7828,51 @@ def read(self, entity=None, attrs=None, ignore=None, params=None):
         return super(System, self).read(entity, attrs, ignore, params)
 
 
+class TailoringFile(
+        Entity,
+        EntityCreateMixin,
+        EntityDeleteMixin,
+        EntityReadMixin,
+        EntitySearchMixin,
+        EntityUpdateMixin):
+    """A representation of a Tailoring File entity."""
+
+    def __init__(self, server_config=None, **kwargs):
+        self._fields = {
+            'name': entity_fields.StringField(
+                required=True,
+                str_type='alpha',
+                length=(4, 30),
+                unique=True
+            ),
+            'scap_file': entity_fields.StringField(),
+            'original_filename': entity_fields.StringField(),
+            'location': entity_fields.OneToManyField(Location),
+            'organization': entity_fields.OneToManyField(Organization),
+        }
+        if 'scap_file' in kwargs:
+            with open(kwargs['scap_file']) as input_file:
+                kwargs['scap_file'] = input_file.read()
+        self._meta = {
+            'api_path': 'api/v2/compliance/tailoring_files',
+            'server_modes': ('sat')
+        }
+        super(TailoringFile, self).__init__(server_config, **kwargs)
+
+    def create_payload(self, **kwargs):
+        """Wrap submitted data within an extra dict."""
+        return {'tailoring_file': super(TailoringFile, self).create_payload()}
+
+    def read(self, entity=None, attrs=None, ignore=None, params=None):
+        """Ignore ``organization``, ``location`` and ``scap_file`` field.
+        """
+
+        if ignore is None:
+            ignore = set()
+        ignore.update(['scap_file', 'location', 'organization'])
```

Comment:
`ignore.update(['scap_file', 'location', 'organization'])`
One thing I want to ask/say about above code: I don't like this way of ignoring fields because when we create these entities only for that time I face that issue but for further read operations we don't need to ignore all the fields.

@SatelliteQE/quality-engineers Do we have any workaround for this problem?

I think we can fix this using overriding `create` method and just returning a json. We don't need to do this for all entities but only for those which we think have important fields and we don't want to ignore those fields. 
```
def create(self, create_missing=None):
        """Override :meth:`nailgun.entity_mixins.EntityCreateMixin.create` because it's causing
        issue of ignoring location and organization fields."""
        return self.create_json(create_missing)
```
I also have one more entity of which I need certain field but can't use it because of this way of ignoring fields.

Reply:
Ignore the above comment, I found how we make it work. Solution was right there but I didn't notice it.